crates/sparrow-instructions/src/evaluators/list/collect_struct.rs

```diff
@@ -0,0 +1,797 @@
+        let input = builder.finish();
+        let input = Arc::new(input);
+
+        let key_indices = UInt32Array::from(vec![0, 0, 0]);
+        let key_capacity = 1;
+
+        let result = CollectStructEvaluator::evaluate_non_windowed(
+            &mut token,
+            key_capacity,
+            &key_indices,
+            input,
+            usize::MAX,
+        )
+        .unwrap();
+        let result = result.as_list::<i32>();
+
+        // build expected result 1
+        let mut builder = ListBuilder::new(StructBuilder::new(fields(), field_builders()));
+        builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(0);
+        builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("a");
+        builder.values().append(true);
+        builder.append(true);
+
+        builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(0);
+        builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("a");
+        builder.values().append(true);
+        builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(1);
+        builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("b");
+        builder.values().append(true);
+        builder.append(true);
+
+        builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(0);
+        builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("a");
+        builder.values().append(true);
+        builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(1);
+        builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("b");
+        builder.values().append(true);
+        builder
+            .values()
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(2);
+        builder
+            .values()
+            .field_builder::<StringBuilder>(1)
+            .unwrap()
+            .append_value("c");
+        builder.values().append(true);
+        builder.append(true);
+        let expected = builder.finish();
+        let expected = Arc::new(expected);
+
+        assert_eq!(expected.as_ref(), result);
+
+        // Batch 2
+        let mut builder = StructBuilder::new(fields(), field_builders());
+        builder
+            .field_builder::<Int64Builder>(0)
+            .unwrap()
+            .append_value(3);
+        builder
```

Comment:
I think this can be cleaned up a bit. What I've done in the past is use a helper that has methods for `field1` and `field2` with the right types. Other patterns exist in the code elsewhere for testing things with structs. Not sure it's necessary. One easy thing would be to allow this test to return a `Result` and then replace `.unwrap()` with `?`.

Reply:
Ah yeah -- I can clean this up big time by constructing each field's column separately and not using the builder.